Nodejs/Product/Nodejs/SharedProject/FileChangeManager.cs

```diff
@@ -227,27 +277,54 @@ internal void IgnoreItemChanges(string fileName, bool ignore)
         /// Stop observing when the file is updated on disk.
         /// </summary>
         /// <param name="fileName">File to stop observing.</param>
-        internal void StopObservingItem(string fileName)
+        public void StopObservingFile(string fileName)
         {
+            this.CheckDisposed();
+
             if (string.IsNullOrEmpty(fileName))
             {
                 throw new ArgumentException(SR.GetString(SR.InvalidParameter), nameof(fileName));
             }
 
             var fullFileName = Utilities.CanonicalizeFileName(fileName);
 
-            if (this.observedItems.ContainsKey(fullFileName))
+            // Remove the file from our observed list. It's important that this is done before the call to 
+            // UnadviseFileChange, because for some reason, the call to UnadviseFileChange can trigger a 
+            // FilesChanged event, and we want to be able to filter that event away.
+            if (this.observedFiles.TryRemove(fullFileName, out var value))
             {
                 // Get the cookie that was used for this.observedItems to this file.
-                var itemInfo = this.observedItems[fullFileName];
+                var (ItemID, FileChangeCookie) = value;
 
-                // Remove the file from our observed list. It's important that this is done before the call to 
-                // UnadviseFileChange, because for some reason, the call to UnadviseFileChange can trigger a 
-                // FilesChanged event, and we want to be able to filter that event away.
-                this.observedItems.Remove(fullFileName);
+                // Stop observing the file
+                ErrorHandler.ThrowOnFailure(this.fileChangeService.UnadviseFileChange(FileChangeCookie));
```

Comment:
>ThrowOnFailure [](start = 29, length = 14)

It looks like `StopObservingFolder` just returns false.  Why the difference?

Reply:
Existing callers required that for `StopObservingFolder`, changed for `*File` since I do prefer that.